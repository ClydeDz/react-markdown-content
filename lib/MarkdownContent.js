"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importStar(require("react"));
const axios_1 = __importDefault(require("axios"));
const react_html_parser_1 = __importDefault(require("react-html-parser"));
const remark_1 = __importDefault(require("remark"));
const remark_html_1 = __importDefault(require("remark-html"));
function MarkdownContent(props) {
    const { content } = props;
    const [appState, setAppState] = react_1.useState("");
    react_1.useEffect(() => {
        axios_1.default.get(content)
            .then(res => {
            remark_1.default()
                .use(remark_html_1.default)
                .process(res.data, function (err, file) {
                const convertedHTML = file.contents;
                setAppState(convertedHTML);
            });
        });
    }, [content]);
    return (react_1.default.createElement(react_1.default.Fragment, null, react_html_parser_1.default(appState)));
}
;
exports.default = MarkdownContent;
