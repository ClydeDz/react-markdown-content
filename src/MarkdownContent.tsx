import React, { useEffect, useState } from "react";
import axios from "axios";
import ReactHtmlParser from "react-html-parser";
import remark from "remark";
import html from "remark-html";
import gfm from "remark-gfm";
import {resolveClassNames} from "./MarkdownContentUtils";

interface IMarkdownContentProps {
    markdownFilePath: string;
    className?: string;
}

export function MarkdownContent(props: IMarkdownContentProps): JSX.Element {
    const { markdownFilePath, className } = props;
    const [appState, setAppState] = useState("");

    useEffect(() => {
        axios.get(markdownFilePath)
        .then(res => {
            remark()
            .use(html)
            .use(gfm)
            .process(res.data, function (err: any, file: any): void {
                const convertedHTML: any = file.contents;
                setAppState(convertedHTML);
            });
        });
      }, [markdownFilePath]);

    return (
        <>
        {appState && (
            <div data-testid="MARKDOWN_CONTENT_CONTAINER" className={resolveClassNames(className)}>
                {ReactHtmlParser(appState)}
            </div>
        )}
        </>
        );
}

