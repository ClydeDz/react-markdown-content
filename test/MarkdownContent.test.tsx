import "@testing-library/jest-dom/extend-expect";
import {act, cleanup, render, screen} from "@testing-library/react";
import axios from "axios";
import React from "react";
import MarkdownContent from "../src/MarkdownContent";
import { GetTestFixture, TestFixtures, TestFixtureResults } from "./fixtures/GetTestFixtures";

jest.mock("axios");
const mockedAxios = axios as jest.Mocked<typeof axios>;
const TEST_ID = "MARKDOWN_CONTENT_CONTAINER";

afterEach(cleanup);

describe("", () => {
    it("should match the snapshot", async () => {
        const res = {data: GetTestFixture(TestFixtures.SIMPLE)};
        mockedAxios.get.mockResolvedValue(res);

        let renderedComponent;
        await act(async () => {
            renderedComponent = render(<MarkdownContent content="" />);
        });

        expect(renderedComponent).toMatchSnapshot();
    });

    it("empty", async () => {
        const res = {data: ""};
        mockedAxios.get.mockResolvedValue(res);

        await act(async () => {
            render(<MarkdownContent content="" />);
        });

        const linkElement = await screen.findByTestId(TEST_ID);
        expect(linkElement.innerHTML).toBe("");
    });

    it.each([
        [TestFixtures.SIMPLE, TestFixtureResults.SIMPLE],
        [TestFixtures.LINKS, TestFixtureResults.LINKS],
    ])("should process markdown from %s and produce correct HTML", async (input, expected) => {
        const res = {data: GetTestFixture(input)};
        mockedAxios.get.mockResolvedValue(res);

        await act(async () => {
            render(<MarkdownContent content="" />);
        });

        const linkElement = await screen.findByTestId(TEST_ID);
        expect(linkElement.innerHTML).toBe(expected);
    });
});
