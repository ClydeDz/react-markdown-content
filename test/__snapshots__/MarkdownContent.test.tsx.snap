// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MarkdownContent /> should match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="markdown-content-container"
        data-testid="MARKDOWN_CONTENT_CONTAINER"
      >
        <h2>
          About
        </h2>
        <p>
          This is a 
          <em>
            simple
          </em>
           
          <strong>
            markdown
          </strong>
           file.
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="markdown-content-container"
      data-testid="MARKDOWN_CONTENT_CONTAINER"
    >
      <h2>
        About
      </h2>
      <p>
        This is a 
        <em>
          simple
        </em>
         
        <strong>
          markdown
        </strong>
         file.
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<MarkdownContent /> should match the snapshot when classname is provided 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="markdown-content-container xyz"
        data-testid="MARKDOWN_CONTENT_CONTAINER"
      >
        <h2>
          About
        </h2>
        <p>
          This is a 
          <em>
            simple
          </em>
           
          <strong>
            markdown
          </strong>
           file.
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="markdown-content-container xyz"
      data-testid="MARKDOWN_CONTENT_CONTAINER"
    >
      <h2>
        About
      </h2>
      <p>
        This is a 
        <em>
          simple
        </em>
         
        <strong>
          markdown
        </strong>
         file.
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<MarkdownContent /> should match the snapshot when no markdown is passed 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      
    </div>
  </body>,
  "container": <div>
    
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
